src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,42 @@
+.calendar {
+  margin: 0 auto;
+  display: flex;
+  flex-wrap: wrap;
+  flex-shrink: 0;
+  gap: 1px;
+  width: $general-size-days + $general-size-bd + (10px * 2);
+  padding: 10px;
+
+  @for $day-of-month from 28 through 31 {
+    &--month-length--#{$day-of-month} > :nth-child(n + #{$day-of-month + 1}) {
+      display: none;
+    }
+  }
+
+  @each $day, $ml in $margin-left {
+    &--start-day--#{$day} > :first-child {
+      margin-left: $ml;
+    }
+  }
+  
+  @for $day from 1 through 31 {
+    & > :nth-child(#{$day}):before {
+      content: "#{$day}";
+    }
+  }
```

Comment:
This should be written under `&__day` selector. No need for unnecessary specificity overload.